aiida/backends/tests/cmdline/commands/test_computer.py

```diff
@@ -459,12 +469,132 @@ def test_computer_test(self):
+        # Exception should be raised
+        self.assertIsNotNone(result.exception)
+
+        # See if the command complains about not getting both names
+        options = ['comp_cli_test_computer']
+        result = self.runner.invoke(computer_rename, options)
+        # Exception should be raised
+        self.assertIsNotNone(result.exception)
+
+        # The new name must be different to the old one
+        options = ['comp_cli_test_computer', 'comp_cli_test_computer']
+        result = self.runner.invoke(computer_rename, options)
+        # Exception should be raised
+        self.assertIsNotNone(result.exception)
+
+        # Change a computer name successully.
+        options = ['comp_cli_test_computer', 'renamed_test_computer']
+        result = self.runner.invoke(computer_rename, options)
+        # Exception should be not be raised
+        self.assertIsNone(result.exception)
+
+        # Check that the name really was changed
+        # The old name should not be available
+        with self.assertRaises(NotExistent):
+            AiidaOrmComputer.get('comp_cli_test_computer')
+        # The new name should be avilable
+        try:
+            AiidaOrmComputer.get('renamed_test_computer')
+        except NotExistent:
+            assertTrue(False)
+
+        # Now change the name back
+        options = ['renamed_test_computer', 'comp_cli_test_computer']
+        result = self.runner.invoke(computer_rename, options)
+        # Exception should be not be raised
+        self.assertIsNone(result.exception)
+
+        # Check that the name really was changed
+        # The old name should not be available
+        with self.assertRaises(NotExistent):
+            AiidaOrmComputer.get('renamed_test_computer')
+        # The new name should be avilable
+        try:
+            AiidaOrmComputer.get('comp_cli_test_computer')
+        except NotExistent:
+            assertTrue(False)
+
+    def test_computer_delete(self):
+        """
+        Test if 'verdi computer delete' command works
+        """
+        from aiida.orm.computer import Computer as AiidaOrmComputer
+        from aiida.common.exceptions import NotExistent
+
+        # See if the command complains about not getting an invalid computer
+        options = ['not_existent_computer_name']
+        result = self.runner.invoke(computer_delete, options)
+        # Exception should be raised
+        self.assertIsNotNone(result.exception)
+
+        # Delete a computer name successully.
+        options = ['comp_cli_test_computer']
+        result = self.runner.invoke(computer_delete, options)
+        # Exception should be not be raised
+        self.assertIsNone(result.exception)
+        # Check that the computer really was deleted
+        with self.assertRaises(NotExistent):
+            AiidaOrmComputer.get('comp_cli_test_computer')
+        # Restore the computer
+        TestVerdiComputerCommands.setUpClass()
```

Comment:
This is sub ideal. If someone changes the `setUpClass` to add another computer for another test in the future, this will try to re-add it and fail. In general the method should be able to rely on being called only once. Consider creating a second computer in `setUpClass` and naming it `computer_for_test_delete` or something like that. Then you can delete that here and not worry.

Reply:
Actually, I don't agree that creating two computers in setUpClass and then deleting it in one of the tests is a good idea. Another test in another class might rely on its presence and, since the order of tests is unspecified, tests might randomly fail.
I would create a `computer_for_test_delete` within the test function itself, and then delete it, and check that indeed it is not there anymore.

Reply:
I agree with Giovanni